### ISSUE TITLE
Android OS 5 Crash

### ISSUE THREAD

**Issue by musternet** [ryanheise/audio_session#32]
The audio_session library version 0.1.3 is causing a crash whenever its being defined in android os 5, downgrading fixed the issue

**Comment by ryanheise** [comment]
Do you have any error message that could help pinpoint the issue?

**Comment by ryanheise** [comment]
I've done a sweep of the code and found a couple of APIs that were not guarded to the SDK level. The latest commit should fix that.

**Comment by musternet** [comment]
> I've done a sweep of the code and found a couple of APIs that were not guarded to the SDK level. The latest commit should fix that.

oh good then, the crash happened in the definition of the AudioSessionPlugin library in the android side anyway
flutterEngine.getPlugins().add(new com.ryanheise.audio_session.AudioSessionPlugin());
i think it might be because of what you mentioned. thanks

**Comment by ryanheise** [comment]
I've just published release 0.1.4 with the above commit. Let me know how it goes, and if you still run into issues.

**Comment by musternet** [comment]
> I've just published release 0.1.4 with the above commit. Let me know how it goes, and if you still run into issues.

unfortunately it didn't work the AudioSessionPlugin can't be instantiated the error being shown in the log is this:

`I/art: Rejecting re-init on previously-failed class java.lang.Class<com.ryanheise.audio_session.AndroidAudioManager$Singleton$1>
I/art: Rejecting re-init on previously-failed class java.lang.Class<com.ryanheise.audio_session.AndroidAudioManager$Singleton$1>
I/art: Rejecting re-init on previously-failed class java.lang.Class<com.ryanheise.audio_session.AndroidAudioManager$Singleton$1>
W/GeneratedPluginsRegister: Tried to automatically register plugins with FlutterEngine (io.flutter.embedding.engine.FlutterEngine@2efb4784) but could not find and invoke the GeneratedPluginRegistrant.
V/FA: Activity resumed, time: 1812723`

**Comment by ryanheise** [comment]
Thanks, that gives me a good lead to go on.

**Comment by ryanheise** [comment]
I've just made another fix attempt, although I haven't published it yet since it might be good to hear whether it works for you.

To try it out from Git `master` you can use this dependency in your `pubspec.yaml`:

```yaml
  audio_session:
    git:
      url: https://github.com/ryanheise/audio_session.git
```

Note that if you use any other plugins that internally use audio_session, you'll also need to add a dependency override to force them to use this version of audio_session:

```yaml
dependency_overrides:
  audio_session:
    git:
      url: https://github.com/ryanheise/audio_session.git
```

Let me know how it goes and if all is good I'll publish another release.

**Comment by musternet** [comment]
I'm using just_audio library that is dependant on audio_session library. I added the dependency override from the git master and everything is working fine now. thanks

**Comment by ryanheise** [comment]
FYI this has now been released in 0.1.5, thanks very much for reporting the issue.
